main.py

```diff
@@ -86,6 +86,11 @@ class MechaClient(Client):
             log.debug("received message from myself ignoring!.")
             return None
 
+        if not str(message).startswith(Commands.prefix):
+            # prevent bot from processing commands without the set prefix
+            log.debug(f"caught a command without prefix. Ignoring.")
+            return None
```

Comment:
Hey, thanks for checking this here now. Raising a special exception in the trigger method isn't necessary any longer, as unknown pointed out. However a small `assert message.startswith(thingy)` in the trigger method wouldn't hurt and be quite cheap. Assertions could even be disabled for production.

One last thing: while it does help with completion in ides, there's no need to explicitly call `str(message)` here, it would only copy the message string, which we don't need.
Python3(.?) does have a syntax for type hinting, much like statically typed languages. In the arguments, `message: str` will enable the ide of your choice (pycharm?) to provide code completion without a runtime hit to performance.

Reply:
Made the changes, thanks for the tips. An assert will definitely point out any funny business with that bit of code.